internal/extract_test.go

```diff
@@ -1,83 +1,151 @@
 package internal_test
 
 import (
+	"bufio"
 	"bytes"
 	"io"
 	"io/ioutil"
+	"os"
+	"strconv"
 	"testing"
 
 	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
 	"github.com/wal-g/wal-g/internal"
 	"github.com/wal-g/wal-g/internal/crypto/openpgp"
 	"github.com/wal-g/wal-g/testtools"
+	"github.com/wal-g/wal-g/utility"
 )
 
 const (
 	PrivateKeyFilePath = "../test/testdata/waleGpgKey"
 )
 
-func TestNoFilesProvided(t *testing.T) {
+func TestExtractAll_noFilesProvided(t *testing.T) {
 	buf := &testtools.NOPTarInterpreter{}
 	err := internal.ExtractAll(buf, []internal.ReaderMaker{})
 	assert.IsType(t, err, internal.NoFilesToExtractError{})
 }
 
-// Tests roundtrip for a tar file.
-func TestTar(t *testing.T) {
-	// Generate and save random bytes compare against compression-decompression cycle.
-	sb := testtools.NewStrideByteReader(10)
+func TestExtractAll_fileDoesntExist(t *testing.T) {
+	readerMaker := &testtools.FileReaderMaker{Key: "testdata/booba.tar"}
+	err := internal.ExtractAll(&testtools.NOPTarInterpreter{}, []internal.ReaderMaker {readerMaker})
+	assert.Error(t, err)
+}
+
+func generateRandomBytes() []byte {
+	sb := testtools.NewStrideByteReader(4)
 	lr := &io.LimitedReader{
 		R: sb,
 		N: int64(1024),
 	}
-	b, err := ioutil.ReadAll(lr)
+	b, _ := ioutil.ReadAll(lr)
+	return b
+}
 
-	// Copy generated bytes to another slice to make the test more robust against modifications of "b".
+func makeTar() (BufferReaderMaker, []byte) {
+	b := generateRandomBytes()
 	bCopy := make([]byte, len(b))
 	copy(bCopy, b)
-	assert.NoError(t, err)
 
-	// Make a tar in memory.
-	member := &bytes.Buffer{}
-	testtools.CreateTar(member, &io.LimitedReader{
-		R: bytes.NewBuffer(b),
-		N: int64(len(b)),
-	})
+	r, w := io.Pipe()
+	go func() {
+		bw := bufio.NewWriterSize(w, 4)
```

Comment:
I suggest to add some comment or extract this to the constant because it's not obvious why size is 4.